src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -376,13 +376,55 @@ internal void FunctionEnvironmentReloadResponse(FunctionEnvironmentReloadRespons
 
             if (res.Result.IsFailure(out Exception reloadEnvironmentVariablesException))
             {
-                _workerChannelLogger.LogError(reloadEnvironmentVariablesException, "Failed to reload environment variables");
-                _reloadTask.SetException(reloadEnvironmentVariablesException);
+                if (IsEnvironmentReloadFailureFatalError(res))
+                {
+                    _workerChannelLogger.LogError(reloadEnvironmentVariablesException, "Failed to reload environment variables");
+                    _reloadTask.SetException(reloadEnvironmentVariablesException);
+                }
+                else
+                {
+                    _reloadTask.SetResult(false);
+                }
+            }
+            else
+            {
+                _reloadTask.SetResult(true);
             }
-            _reloadTask.SetResult(true);
             latencyEvent.Dispose();
         }
 
+        /// <summary>
+        /// Determines whether the environment reload failure should be considered as a fatal error or should be ignored.
+        /// </summary>
+        internal bool IsEnvironmentReloadFailureFatalError(FunctionEnvironmentReloadResponse response)
+        {
+            var workerRuntime = _environment.GetFunctionsWorkerRuntime();
+
+            // Currently we support non fatal environment reload errors for dotnet isolated worker only.
+            if (workerRuntime != RpcWorkerConstants.DotNetIsolatedLanguageWorkerName)
+            {
+                return true;
+            }
+
+            var rpcException = response.Result.Exception;
+
+            // Currently the below 2 errors are considered as non fatal.
+            if (rpcException.Message.StartsWith("Microsoft.Azure.Functions.Worker.EnvironmentReloadNotSupportedException"))
```

Comment:
The native host will include this specific exception if the app payload uses an older version of worker package which does not support placeholders. https://github.com/Azure/azure-functions-dotnet-worker/blob/a70ead9e497c990d6c29b920f7ed1c3044f6c2f7/host/src/FunctionsNetHost/Grpc/IncomingGrpcMessageHandler.cs#L64-L67

Reply:
We should have this exception message be something that isn't specific to .NET. 

Ideally, this would be a flow any worker (any stack) can use.